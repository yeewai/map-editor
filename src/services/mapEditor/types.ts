export interface State {
    zoom: number,
    panV: number,
    panH: number
}
