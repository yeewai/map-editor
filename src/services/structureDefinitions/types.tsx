export interface StructureDefinition {
    id: string,
    name: string,
    description?: string,
    width: number,
    length: number,
    kind: string,
    imageUrl: string,
    imageWidth?: number,
    imageHeight?: number,
    image?: any
}

export const emptyStructureDefintion = {
    id: "",
    name: "",
    width: 1,
    length: 1,
    kind: "",
    imageUrl: ""
}

export interface State {
    items: StructureDefinition[],
    itemsRequested: number,
    isFetching: boolean,
    hasFetched: boolean,
    error?: string
}
