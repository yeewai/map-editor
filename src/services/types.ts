import { State as structureDefinitionType } from './structureDefinitions/types';
import { State as worldType } from './worlds/types';

import { defaultState as structureDefinitionState } from './structureDefinitions/reducer';
import { defaultState as worldState } from './worlds/reducer';

export interface StateTree {
    structureDefinitions: structureDefinitionType,
    worlds: worldType
}

export const defaultStateTree: StateTree = {
    structureDefinitions: structureDefinitionState,
    worlds: worldState
}

export interface Action {
    type: string,
    payload?: any
}

export const defaultAction:Action = {
    type: ""
}
