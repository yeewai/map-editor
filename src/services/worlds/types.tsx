import { structureDefinitionTypes } from 'services/structureDefinitions';

export interface Structure {
    name: string,
    description?: string,
    definitionId: string,
    definition?: structureDefinitionTypes.StructureDefinition
    xposition: number,
    yposition: number
}

export const emptyStructure = {
    name: "",
    definitionId: structureDefinitionTypes.emptyStructureDefintion.id,
    definition: structureDefinitionTypes.emptyStructureDefintion,
    xposition: 1,
    yposition: 1
}

export interface World {
    id: string,
    name: string,
    key: string,
    description?: string,
    width: number,
    length: number,
    createdAt: number,
    saveMessage?: string,
    isPublished: boolean,
    structures?: Structure[],
    nullStructureId: string
}

export const emptyWorld = {
    id: "",
    name: "",
    key: "",
    width: 1,
    length: 1,
    createdAt: 1,
    isPublished: false,
    structures: [ emptyStructure ],
    nullStructureId: structureDefinitionTypes.emptyStructureDefintion.id
}

export interface WorldWithBoard extends World {
    board: Structure[][],
    uniqueStructures: structureDefinitionTypes.StructureDefinition[]
}

export const emptyWorldWithBoard = {
    ...emptyWorld,
    board: [
        [ emptyStructure ]
    ],
    uniqueStructures: [ structureDefinitionTypes.emptyStructureDefintion ]
}

export interface State {
    items: World[],
    isFetching: boolean,
    hasFetched: boolean,
    error?: string
}
