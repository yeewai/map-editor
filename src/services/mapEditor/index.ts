import reducer from "./reducer";
import * as mapEditorActions from "./actions";
import * as mapEditorSelectors from "./selectors";
import * as mapEditorTypes from "./types";

export { mapEditorActions, mapEditorSelectors, mapEditorTypes };
export default reducer;
