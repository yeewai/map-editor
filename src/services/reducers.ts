import { routerReducer } from 'react-router-redux';
import { combineReducers } from 'redux';
import { modalReducer } from 'ever-modal';
import { reducer as formReducer } from 'redux-form'

import { StateTree } from './types';
import structureDefinitions from './structureDefinitions';
import worlds from './worlds';

const rootReducer = combineReducers<StateTree>({
    router: routerReducer,
    modals: modalReducer,
    form: formReducer,
    structureDefinitions,
    worlds

});

export default rootReducer;
