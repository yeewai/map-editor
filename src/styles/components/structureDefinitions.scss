.card-li {
    // cursor: pointer;
    // // background: var(--svp-content-bg);
    // border: none;
    // color: black;//var(--svp-body-text);
    // max-width: 180px;
    // height: 180px;
    // max-height: 180px;
    margin: 10px;
    // text-align: center;
    display: inline-block;

    .card {
        width: 240px;
        height: 360px;
        box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
        transition: box-shadow 0.4s cubic-bezier(.25,.8,.25,1);
        border: none;

        &:hover, &:focus {
            box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
            color: var(--svp-body-text);
            text-decoration: none;
            outline: none;
        }

        img {
            height: 160px;
        }

        .card-title, .card-subtitle {
            font-weight: 300;
        }

        .card-subtitle {
            font-size: .8rem;
        }
    }

}
