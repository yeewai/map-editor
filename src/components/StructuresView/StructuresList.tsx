import * as React from 'react';
import { connect } from "react-redux";
import { Card, CardImg, CardText, CardBody, CardTitle, CardSubtitle, Button,
     ListGroup, ListGroupItem } from 'reactstrap';

import { OpenModalButton } from 'ever-modal';

import { StateTree } from 'services/types';

import _ from 'lodash';
import { structureDefinitionsSelectors, structureDefinitionTypes } from 'services/structureDefinitions';

export interface StateProps {
    structureDefinitions: _.Dictionary<structureDefinitionTypes.StructureDefinition[]>
}

export const StructuresList: React.SFC<StateProps> = ( props ) => {
    const { structureDefinitions } = props;

    return (
        <ListGroup>
            <ListGroupItem key="title">
                <h2>Structure Library</h2>
                <OpenModalButton modalType="ADD_STRUCTURE_DEFINITION" modalProps={{ariaLabel: "Sup", formName: "structureDefinition"}} >New</OpenModalButton>
            </ListGroupItem>
            {
                Object.keys(structureDefinitions).map( (key) => (
                    <ListGroupItem key={key}>
                        <h3>{key}</h3>
                        <ul>
                            { structureDefinitions[key].map( ( sd: structureDefinitionTypes.StructureDefinition ) => (
                                <li className="card-li" key={sd.id} >
                                    <Card>
                                        <CardImg top width="100%" src={sd.imageUrl} alt={sd.name} />
                                        <CardBody>
                                            <CardTitle>{sd.name}</CardTitle>
                                            <CardSubtitle>{sd.width} x {sd.length}</CardSubtitle>
                                            <div className="card-text" dangerouslySetInnerHTML={{ __html: sd.description || "" }} />
                                            <OpenModalButton modalType="EDIT_STRUCTURE_DEFINITION" modalProps={{ariaLabel: "Sup", sd: sd, formName: "structureDefinition"}} >Edit</OpenModalButton>
                                        </CardBody>
                                    </Card>
                                </li>
                            )) }
                        </ul>
                    </ListGroupItem>
                ))
            }
        </ListGroup>
    );
}

export const mapStateToProps = (state: StateTree) => ({
    structureDefinitions: structureDefinitionsSelectors.getByGroup(state)
});


export default connect<StateProps, any, any>(mapStateToProps)(StructuresList);
