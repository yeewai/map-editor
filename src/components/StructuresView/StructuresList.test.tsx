import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import $ from 'jquery';
/*import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';*/
/*import sinon from 'sinon';*/

import { StructuresList, mapStateToProps, StateProps } from './StructuresList';
import { StateTree, defaultStateTree } from 'services/types';

import { structureDefinitionTypes } from 'services/structureDefinitions';

Enzyme.configure({ adapter: new Adapter() });

describe("Structures List", () => {
    const sd1: structureDefinitionTypes.StructureDefinition = {
        "id": "id1",
        "name": "name1",
        "description": "string",
        "width": 1,
        "length": 2,
        "kind": "someKind",
        "imageUrl": "string"
    };
    const sd2: structureDefinitionTypes.StructureDefinition = {
        "id": "id2",
        "name": "name2",
        "description": "string",
        "width": 2,
        "length": 5,
        "kind": "someKind",
        "imageUrl": "string"
    };
    const sd3: structureDefinitionTypes.StructureDefinition = {
        "id": "id3",
        "name": "name3",
        "description": undefined,
        "width": 3,
        "length": 4,
        "kind": "someOtherKind",
        "imageUrl": "string"
    };

    const sd = {
        someKind: [ sd1, sd2],
        someOtherKind: [ sd3 ]
    };

    it( "matches snapshot", () => {
        const component = () => (<div />);
        const wrapper = shallow(<StructuresList structureDefinitions={sd} LiComponent={component}/>);
        expect(wrapper).toMatchSnapshot();
    })

    it( "maps state to props", () => {
        const state: StateTree = $.extend(true, {}, defaultStateTree);
        state.structureDefinitions.items = [ sd1, sd2, sd3 ];

        expect(mapStateToProps(state)).toEqual({structureDefinitions: sd});
    })
})
