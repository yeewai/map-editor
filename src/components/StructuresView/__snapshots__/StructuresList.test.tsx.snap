// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structures List matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    LiComponent={[Function]}
    structureDefinitions={
        Object {
            "someKind": Array [
              Object {
                "description": "string",
                "id": "id1",
                "imageUrl": "string",
                "kind": "someKind",
                "length": 2,
                "name": "name1",
                "width": 1,
              },
              Object {
                "description": "string",
                "id": "id2",
                "imageUrl": "string",
                "kind": "someKind",
                "length": 5,
                "name": "name2",
                "width": 2,
              },
            ],
            "someOtherKind": Array [
              Object {
                "description": undefined,
                "id": "id3",
                "imageUrl": "string",
                "kind": "someOtherKind",
                "length": 4,
                "name": "name3",
                "width": 3,
              },
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ListGroupItem
          tag="li"
>
          <h2>
                    Structure Library
          </h2>
          <Connect(OpenModalButton)
                    modalProps={
                              Object {
                                        "ariaLabel": "Sup",
                                        "formName": "structureDefinition",
                                      }
                    }
                    modalType="ADD_STRUCTURE_DEFINITION"
          >
                    New
          </Connect(OpenModalButton)>
</ListGroupItem>,
        Array [
          <ListGroupItem
            tag="li"
>
            <h3>
                        someKind
            </h3>
            <ul>
                        <component
                                    structureDefinition={
                                                Object {
                                                            "description": "string",
                                                            "id": "id1",
                                                            "imageUrl": "string",
                                                            "kind": "someKind",
                                                            "length": 2,
                                                            "name": "name1",
                                                            "width": 1,
                                                          }
                                    }
                        />
                        <component
                                    structureDefinition={
                                                Object {
                                                            "description": "string",
                                                            "id": "id2",
                                                            "imageUrl": "string",
                                                            "kind": "someKind",
                                                            "length": 5,
                                                            "name": "name2",
                                                            "width": 2,
                                                          }
                                    }
                        />
            </ul>
</ListGroupItem>,
          <ListGroupItem
            tag="li"
>
            <h3>
                        someOtherKind
            </h3>
            <ul>
                        <component
                                    structureDefinition={
                                                Object {
                                                            "description": undefined,
                                                            "id": "id3",
                                                            "imageUrl": "string",
                                                            "kind": "someOtherKind",
                                                            "length": 4,
                                                            "name": "name3",
                                                            "width": 3,
                                                          }
                                    }
                        />
            </ul>
</ListGroupItem>,
        ],
      ],
      "className": "structure-library",
      "tag": "ul",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "title",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <h2>
              Structure Library
</h2>,
            <Connect(OpenModalButton)
              modalProps={
                            Object {
                                          "ariaLabel": "Sup",
                                          "formName": "structureDefinition",
                                        }
              }
              modalType="ADD_STRUCTURE_DEFINITION"
>
              New
</Connect(OpenModalButton)>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Structure Library",
            },
            "ref": null,
            "rendered": "Structure Library",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "New",
              "modalProps": Object {
                "ariaLabel": "Sup",
                "formName": "structureDefinition",
              },
              "modalType": "ADD_STRUCTURE_DEFINITION",
            },
            "ref": null,
            "rendered": "New",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "someKind",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <h3>
              someKind
</h3>,
            <ul>
              <component
                            structureDefinition={
                                          Object {
                                                        "description": "string",
                                                        "id": "id1",
                                                        "imageUrl": "string",
                                                        "kind": "someKind",
                                                        "length": 2,
                                                        "name": "name1",
                                                        "width": 1,
                                                      }
                            }
              />
              <component
                            structureDefinition={
                                          Object {
                                                        "description": "string",
                                                        "id": "id2",
                                                        "imageUrl": "string",
                                                        "kind": "someKind",
                                                        "length": 5,
                                                        "name": "name2",
                                                        "width": 2,
                                                      }
                            }
              />
</ul>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "someKind",
            },
            "ref": null,
            "rendered": "someKind",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <component
                  structureDefinition={
                                    Object {
                                                      "description": "string",
                                                      "id": "id1",
                                                      "imageUrl": "string",
                                                      "kind": "someKind",
                                                      "length": 2,
                                                      "name": "name1",
                                                      "width": 1,
                                                    }
                  }
/>,
                <component
                  structureDefinition={
                                    Object {
                                                      "description": "string",
                                                      "id": "id2",
                                                      "imageUrl": "string",
                                                      "kind": "someKind",
                                                      "length": 5,
                                                      "name": "name2",
                                                      "width": 2,
                                                    }
                  }
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "id1",
                "nodeType": "function",
                "props": Object {
                  "structureDefinition": Object {
                    "description": "string",
                    "id": "id1",
                    "imageUrl": "string",
                    "kind": "someKind",
                    "length": 2,
                    "name": "name1",
                    "width": 1,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "id2",
                "nodeType": "function",
                "props": Object {
                  "structureDefinition": Object {
                    "description": "string",
                    "id": "id2",
                    "imageUrl": "string",
                    "kind": "someKind",
                    "length": 5,
                    "name": "name2",
                    "width": 2,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "someOtherKind",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <h3>
              someOtherKind
</h3>,
            <ul>
              <component
                            structureDefinition={
                                          Object {
                                                        "description": undefined,
                                                        "id": "id3",
                                                        "imageUrl": "string",
                                                        "kind": "someOtherKind",
                                                        "length": 4,
                                                        "name": "name3",
                                                        "width": 3,
                                                      }
                            }
              />
</ul>,
          ],
          "tag": "li",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "someOtherKind",
            },
            "ref": null,
            "rendered": "someOtherKind",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <component
                  structureDefinition={
                                    Object {
                                                      "description": undefined,
                                                      "id": "id3",
                                                      "imageUrl": "string",
                                                      "kind": "someOtherKind",
                                                      "length": 4,
                                                      "name": "name3",
                                                      "width": 3,
                                                    }
                  }
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "id3",
                "nodeType": "function",
                "props": Object {
                  "structureDefinition": Object {
                    "description": undefined,
                    "id": "id3",
                    "imageUrl": "string",
                    "kind": "someOtherKind",
                    "length": 4,
                    "name": "name3",
                    "width": 3,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListGroupItem
            tag="li"
>
            <h2>
                        Structure Library
            </h2>
            <Connect(OpenModalButton)
                        modalProps={
                                    Object {
                                                "ariaLabel": "Sup",
                                                "formName": "structureDefinition",
                                              }
                        }
                        modalType="ADD_STRUCTURE_DEFINITION"
            >
                        New
            </Connect(OpenModalButton)>
</ListGroupItem>,
          Array [
            <ListGroupItem
              tag="li"
>
              <h3>
                            someKind
              </h3>
              <ul>
                            <component
                                          structureDefinition={
                                                        Object {
                                                                      "description": "string",
                                                                      "id": "id1",
                                                                      "imageUrl": "string",
                                                                      "kind": "someKind",
                                                                      "length": 2,
                                                                      "name": "name1",
                                                                      "width": 1,
                                                                    }
                                          }
                            />
                            <component
                                          structureDefinition={
                                                        Object {
                                                                      "description": "string",
                                                                      "id": "id2",
                                                                      "imageUrl": "string",
                                                                      "kind": "someKind",
                                                                      "length": 5,
                                                                      "name": "name2",
                                                                      "width": 2,
                                                                    }
                                          }
                            />
              </ul>
</ListGroupItem>,
            <ListGroupItem
              tag="li"
>
              <h3>
                            someOtherKind
              </h3>
              <ul>
                            <component
                                          structureDefinition={
                                                        Object {
                                                                      "description": undefined,
                                                                      "id": "id3",
                                                                      "imageUrl": "string",
                                                                      "kind": "someOtherKind",
                                                                      "length": 4,
                                                                      "name": "name3",
                                                                      "width": 3,
                                                                    }
                                          }
                            />
              </ul>
</ListGroupItem>,
          ],
        ],
        "className": "structure-library",
        "tag": "ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "title",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h2>
                Structure Library
</h2>,
              <Connect(OpenModalButton)
                modalProps={
                                Object {
                                                "ariaLabel": "Sup",
                                                "formName": "structureDefinition",
                                              }
                }
                modalType="ADD_STRUCTURE_DEFINITION"
>
                New
</Connect(OpenModalButton)>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Structure Library",
              },
              "ref": null,
              "rendered": "Structure Library",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "New",
                "modalProps": Object {
                  "ariaLabel": "Sup",
                  "formName": "structureDefinition",
                },
                "modalType": "ADD_STRUCTURE_DEFINITION",
              },
              "ref": null,
              "rendered": "New",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "someKind",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h3>
                someKind
</h3>,
              <ul>
                <component
                                structureDefinition={
                                                Object {
                                                                "description": "string",
                                                                "id": "id1",
                                                                "imageUrl": "string",
                                                                "kind": "someKind",
                                                                "length": 2,
                                                                "name": "name1",
                                                                "width": 1,
                                                              }
                                }
                />
                <component
                                structureDefinition={
                                                Object {
                                                                "description": "string",
                                                                "id": "id2",
                                                                "imageUrl": "string",
                                                                "kind": "someKind",
                                                                "length": 5,
                                                                "name": "name2",
                                                                "width": 2,
                                                              }
                                }
                />
</ul>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "someKind",
              },
              "ref": null,
              "rendered": "someKind",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <component
                    structureDefinition={
                                        Object {
                                                            "description": "string",
                                                            "id": "id1",
                                                            "imageUrl": "string",
                                                            "kind": "someKind",
                                                            "length": 2,
                                                            "name": "name1",
                                                            "width": 1,
                                                          }
                    }
/>,
                  <component
                    structureDefinition={
                                        Object {
                                                            "description": "string",
                                                            "id": "id2",
                                                            "imageUrl": "string",
                                                            "kind": "someKind",
                                                            "length": 5,
                                                            "name": "name2",
                                                            "width": 2,
                                                          }
                    }
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "id1",
                  "nodeType": "function",
                  "props": Object {
                    "structureDefinition": Object {
                      "description": "string",
                      "id": "id1",
                      "imageUrl": "string",
                      "kind": "someKind",
                      "length": 2,
                      "name": "name1",
                      "width": 1,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "id2",
                  "nodeType": "function",
                  "props": Object {
                    "structureDefinition": Object {
                      "description": "string",
                      "id": "id2",
                      "imageUrl": "string",
                      "kind": "someKind",
                      "length": 5,
                      "name": "name2",
                      "width": 2,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "someOtherKind",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h3>
                someOtherKind
</h3>,
              <ul>
                <component
                                structureDefinition={
                                                Object {
                                                                "description": undefined,
                                                                "id": "id3",
                                                                "imageUrl": "string",
                                                                "kind": "someOtherKind",
                                                                "length": 4,
                                                                "name": "name3",
                                                                "width": 3,
                                                              }
                                }
                />
</ul>,
            ],
            "tag": "li",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "someOtherKind",
              },
              "ref": null,
              "rendered": "someOtherKind",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <component
                    structureDefinition={
                                        Object {
                                                            "description": undefined,
                                                            "id": "id3",
                                                            "imageUrl": "string",
                                                            "kind": "someOtherKind",
                                                            "length": 4,
                                                            "name": "name3",
                                                            "width": 3,
                                                          }
                    }
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "id3",
                  "nodeType": "function",
                  "props": Object {
                    "structureDefinition": Object {
                      "description": undefined,
                      "id": "id3",
                      "imageUrl": "string",
                      "kind": "someOtherKind",
                      "length": 4,
                      "name": "name3",
                      "width": 3,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
